mixer.cpp

```diff
@@ -103,16 +155,21 @@ struct Mixer_impl {
 			const int rate = READ_LE_UINT32(fmt + 4);
 			const int bits = READ_LE_UINT16(fmt + 14);
 			debug(DBG_SND, "wave format %d channels %d rate %d bits %d", format, channels, rate, bits);
-			const bool isMixerFormat = (format == 1 && channels == 2 && rate == kMixFreq && bits == 16);
-			if (isMixerFormat && (AUDIO_S16SYS == AUDIO_S16LSB)) {
+			const bool isMixerFormat = (format == 1 && channels == kMixSoundChannels && rate == kMixFreq && bits == SDL_AUDIO_BITSIZE(kMixFormat) && !SDL_AUDIO_ISFLOAT(kMixFormat));
+			if (freq == 0 && isMixerFormat && (AUDIO_S16SYS == AUDIO_S16LSB)) {
 				Mix_Chunk *chunk = Mix_QuickLoad_WAV(const_cast<uint8_t *>(data));
 				playSound(channel, volume, chunk, loops);
 				return;
 			}
 		}
 		const uint32_t size = READ_LE_UINT32(data + 4) + 8;
 		SDL_RWops *rw = SDL_RWFromConstMem(data, size);
-		Mix_Chunk *chunk = Mix_LoadWAV_RW(rw, 1);
+		Mix_Chunk *chunk = 0;
+		if (freq == 0) {
```

Comment:
Is that condition still reachable?

Reply:
Not at the current state. And the same applies to the condition a few lines up (where Mix_QuickLoad_WAV is called). Both can be removed, but do you want to keep reading information from .wav file (when it's used only to debug) or remove it ?

Reply:
I think it makes the code simpler to remove the two conditions as they are unused.